File: app/src/main/java/the/mrsmile/foodorderer/adapters/ViewPagerAdapter.java
package the.mrsmile.foodorderer.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;
import the.mrsmile.foodorderer.R;

public class ViewPagerAdapter extends RecyclerView.Adapter<ViewPagerAdapter.ViewHolder> {

    List<Integer> list;
    OnClickInterface listener;

    public ViewPagerAdapter(List<Integer> list, OnClickInterface listener) {
        this.list = list;
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_pager_items, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        int currentItem = list.get(position);
        holder.image.setImageResource(currentItem);

    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        public ImageView image;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            image = itemView.findViewById(R.id.ivViewPager);

//            image.setClipToOutline(true);
        }

        //init block
        {
            itemView.setOnClickListener(this);
        }


        //implemented method of View.OnClickListener interface
        @Override
        public void onClick(View view) {
            listener.onCategoriesClick(getAdapterPosition());
        }
    }

    //created a new interface to control on Click events from main activity
    public interface OnClickInterface {
        void onCategoriesClick(int position);
    }
}

File: app/src/main/java/the/mrsmile/foodorderer/adapters/ViewPagerAdapter.kt
package the.mrsmile.foodorderer.adapters

import android.content.Context
import android.util.Log
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.ProgressBar
import androidx.swiperefreshlayout.widget.CircularProgressDrawable
import com.bumptech.glide.Glide
import com.squareup.picasso.Callback
import com.squareup.picasso.Picasso
import the.mrsmile.foodorderer.R
import the.mrsmile.foodorderer.activities.MainActivity
import the.mrsmile.foodorderer.models.CategoryItems
import java.lang.Exception

class ViewPagerAdapter(private var list: List<CategoryItems>, var listener: OnClickInterface,val context: Context) :
    RecyclerView.Adapter<ViewPagerAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.view_pager_items, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val currentItem = list[position]
        if (!list[position].image.isNullOrEmpty())
            Glide.with(context).load(currentItem.image).placeholder(R.drawable.placeholder2).centerCrop().into(holder.image)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView),
        View.OnClickListener {

        var image: ImageView = itemView.findViewById(R.id.ivViewPager)

        //implemented method of View.OnClickListener interface
        override fun onClick(view: View) {
            listener.onCategoriesClick(absoluteAdapterPosition)
        }

        //init block
        init {
            itemView.setOnClickListener(this)
        }
    }

    //created a new interface to control on Click events from main activity
    interface OnClickInterface {
        fun onCategoriesClick(position: Int)
    }
}